kratos/utilities/binbased_fast_point_locator.h

```diff
@@ -67,24 +67,25 @@ namespace Kratos
  * @author  Riccardo Rossi <rrossi@cimne.upc.edu>
  * @note The location function is threadsafe, and can be used in OpenMP loops
  * @tparam TDim If we work in a 2D or 3D space
- * @tparam ConfigureType The spatial container
+ * @tparam TConfigureType The spatial container
  */
-template< SizeType TDim, class ConfigureType = SpatialContainersConfigure<TDim> >
+template< SizeType TDim, class TConfigureType = SpatialContainersConfigure<TDim> >
```

Comment:
can we do it like this 
`template< SizeType TDim, class TEntity = Element, class TConfigureType = SpatialContainersConfigure<TDim, TEntity> >`

Reply:
This can simplify the declaration of object like : 
auto point_locator = BinBasedFastPointLocator<3, Condition>(this_model_part);

instead of 

auto point_locator = BinBasedFastPointLocator<3, SpatialContainersConfigure<3, Condition>>(this_model_part);

Reply:
I have another idea, I am not very friend of duplicate (redundant) unnecessarily the template arguments (more compilation type)

Reply:
I created BinBasedFastPointLocatorConditions deriring from BinBasedFastPointLocator, then it will be easy to use and not add new layers of template, that additionally breaks the retocompatibility

Reply:
That is fine. But then can we put this in a different file and expose it to python too ?

Reply:
putting it in a different file make it clean and clear about the contents of the file (according to the name) ...

Reply:
Done